### PR TITLE
chore: Aries JS API Documentation

### LINKED ISSUE #1356: Aries JS API Documentation
Document JS Aries APIs.

### PR DESCRIPTION
**Pull request by rolsonquadras**
closes #1356 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: cmd/aries-js-worker/src/aries.js [hunk 1]
```diff
@@ -176,14 +176,40 @@ const Aries = function(opts) {
...
+         * DIDExchange methods - Refer to [OpenAPI spec](docs/rest/openapi_spec.md#generate-openapi-spec) for
+         * input params and output return json values.
+         */
         didexchange: {
             pkgname: "didexchange",
+
+            /**
+             * Creates a DID Exchange Invitation.
+             *
+             * @param text - json document
```
Comment (troyronda) [inline cmd/aries-js-worker/src/aries.js]: `text` is an odd argument name. should we fix?

Comment (rolsonquadras) [reply cmd/aries-js-worker/src/aries.js]: [Updated](https://github.com/hyperledger/aries-framework-go/compare/e1b402f9989822b55eebe082c9a4acf49252fba1..e02e4a41644e17e908307b9dbfbe7b1710b94960#diff-a43de5ddb6dcf66800367a9232d43f90R189) to name of the existing param to req.

File: cmd/aries-js-worker/src/aries.js [hunk 2]
```diff
@@ -176,90 +176,255 @@ const Aries = function(opts) {
...
+         */
         didexchange: {
             pkgname: "didexchange",
-            createInvitation: async function (text) {
+
+            /**
+             * Creates a DID Exchange Invitation.
+             *
+             * @param req - json document
+             * @returns {Promise<String>}
```
Comment (troyronda) [inline cmd/aries-js-worker/src/aries.js]: I guess there isn't a good way to indicate that the Promise is for the JSON response from the Controller?

File: cmd/aries-js-worker/src/aries.js [hunk 3]
```diff
@@ -176,90 +176,255 @@ const Aries = function(opts) {
...
+            /**
+             * Accepts a DID Exchange invitation.
+             *
+             * @param req - json document
+             * @returns {Promise<String>}
+             */
+            acceptInvitation: async function (req) {
                 return new Promise((resolve, reject) => {
-                    invoke(aw, pending,  this.pkgname, "AcceptInvitation", {id: connectionID, public: publicDID}, "timeout while accepting invitation").then(
+                    invoke(aw, pending,  this.pkgname, "AcceptInvitation", req, "timeout while accepting invitation").then(
```
Comment (rolsonquadras) [inline cmd/aries-js-worker/src/aries.js]: Reverted API param changes done in https://github.com/hyperledger/aries-framework-go/pull/1357/files#diff-a43de5ddb6dcf66800367a9232d43f90R187 as index.html doesnt support multiple params. Need to revisit this after v0.1.2 version.

File: cmd/aries-js-worker/src/aries.js [hunk 4]
```diff
@@ -176,90 +176,255 @@ const Aries = function(opts) {
...
+             *
+             * @returns {Promise<String>}
+             */
             getConnection: async function () {
                 return invoke(aw, pending,  this.pkgname, "GetConnection", "{}", "timeout while fetching router connection id")
             }
         },
 
+        /**
+         * Verifiable methods related to credentials and presentations - Refer to [OpenAPI spec](docs/rest/openapi_spec.md#generate-openapi-spec) for
```
Comment (troyronda) [inline cmd/aries-js-worker/src/aries.js]: TODO: should these functions be under vdri? (it's a bigger question than this PR).

File: cmd/aries-js-worker/src/aries.js (hunk 3 above)
Comment (troyronda) [reply cmd/aries-js-worker/src/aries.js]: Issue number?

Comment (rolsonquadras) [reply cmd/aries-js-worker/src/aries.js]: https://github.com/hyperledger/aries-framework-go/issues/1366

File: cmd/aries-js-worker/src/aries.js (hunk 4 above)
Comment (rolsonquadras) [reply cmd/aries-js-worker/src/aries.js]: Created an issue https://github.com/hyperledger/aries-framework-go/issues/1367.

File: cmd/aries-js-worker/src/aries.js (hunk 2 above)
Comment (llorllale) [reply cmd/aries-js-worker/src/aries.js]: Should instead be `Promise<Object>~? Unless we start defining specific types for the models.... (nah)

File: cmd/aries-js-worker/src/aries.js [hunk 5]
```diff
@@ -176,90 +176,255 @@ const Aries = function(opts) {
...
+                    invoke(aw, pending,  this.pkgname, "AcceptInvitation", req, "timeout while accepting invitation").then(
                         resp => resolve(resp),
                         err => reject(new Error("failed to accept invitation: " + err.message))
                     )
                 })
             },
-            acceptExchangeRequest: async function (text) {
+
+            /**
+             * Accepts a DID Exchange invitation.
```
Comment (llorllale) [inline cmd/aries-js-worker/src/aries.js]: > Accepts a DID Exchange request

File: cmd/aries-js-worker/src/aries.js [hunk 6]
```diff
@@ -176,90 +176,255 @@ const Aries = function(opts) {
...
             },
-            createImplicitInvitation: async function (text) {
+
+            /**
+             * Creates an implicit invitation using inviter DID.
+             *
+             * @param req - json document
+             * @returns {Promise<String>}
+             */
+            createImplicitInvitation: async function (req) {
```
Comment (llorllale) [inline cmd/aries-js-worker/src/aries.js]: rename to `req` not propagated to call to `invoke()`

File: cmd/aries-js-worker/src/aries.js (hunk 5 above)
Comment (rolsonquadras) [reply cmd/aries-js-worker/src/aries.js]: [Updated](https://github.com/hyperledger/aries-framework-go/compare/415b0b2413e6b4948632dec5a47ce53406011239..eeb7de2aa3ffaab86cf1542c90f7c65569c8fd2c#diff-a43de5ddb6dcf66800367a9232d43f90R222)

File: cmd/aries-js-worker/src/aries.js (hunk 2 above)
Comment (rolsonquadras) [reply cmd/aries-js-worker/src/aries.js]: [Updated](https://github.com/hyperledger/aries-framework-go/compare/415b0b2413e6b4948632dec5a47ce53406011239..eeb7de2aa3ffaab86cf1542c90f7c65569c8fd2c#diff-a43de5ddb6dcf66800367a9232d43f90R123) to `Promise<Object>`

File: cmd/aries-js-worker/src/aries.js (hunk 6 above)
Comment (rolsonquadras) [reply cmd/aries-js-worker/src/aries.js]: [Done](https://github.com/hyperledger/aries-framework-go/compare/415b0b2413e6b4948632dec5a47ce53406011239..eeb7de2aa3ffaab86cf1542c90f7c65569c8fd2c#diff-a43de5ddb6dcf66800367a9232d43f90R238).

**Pull request merged**